.github/workflows/commitlint.yml

```diff
@@ -0,0 +1,15 @@
+name: Lint Commit Messages
+on:
+  push:
+    branches: [main]
```

Comment:
**Nitpick**

Do we need to lint when pushing to main? Or only on PR?

Reply:
This will lint both on main and PR.  We need to lint on PRs. I'm not sure about main. What do you think?

Reply:
My vote is only on PRs, not on main.

Reply:
fine with me. I'll update